src/Store.ts

```diff
@@ -333,6 +340,14 @@ export const history: Reducer<HistoryState> = (
                 selectedActivity: action.selectedActivity
             };
 
+        case 'Clicked_SuggestedActions':
+
+            action.clickedActivity.clicked = true;
+
+            return {
+                ... state
+            };
+
```

Comment:
No no no no no no no no no no no no no no no - the first rule of Redux Club is you don't mutate state. It's also the second rule. (Yes, I did just tweet that.) We need to generate a new activity here, and also update `activities` and maybe `selectedActivity`. The closest model is `'Send_Message_Succeed'`. I think this is right:

```typescript
        case 'Take_SuggestedAction': {
            const i = state.activities.findIndex(activity => activity === action.trackedMessage);
            const activity = state.activities[i];
            const newActivity = {
                ... activity,
                suggestedActionTaken: true
            };
            return {
                ... state,
                activities: [
                    ... state.activities.slice(0, i),
                    newActivity,
                    ... state.activities.slice(i + 1)
                ],
                selectedActivity: state.selectedActivity === activity ? newActivity : state.selectedActivity
            }
```

You'll also need to add `'Take_SuggestedAction'` to the list of action types in `updateSelectedActivity` below.

Yeah, it's a lot more work but it's an important rule.

(I could imagine adding a helper function for the array manipulation since we do it in a few places in this reducer.)